addons/point_of_sale/models/pos_session.py

```diff
@@ -313,6 +302,522 @@ def action_pos_session_close(self):
+        }]
+
+    @api.model
+    def _credit_amounts(self, partial_move_line_vals, amount, amount_converted, session_currency, is_using_company_currency):
+        """ `partial_move_line_vals` is completed by `credit`ing the given amounts.
+
+        NOTE In pos, all passed amounts are in the currency of journal_id in the session.config_id.
+            This means that amount fields in any pos record are actually equivalent to amount_currency
+            in account module. Understanding this basic is important in correctly assigning values for
+            'amount' and 'amount_currency' in the account.move.line record.
+
+        :param partial_move_line_vals dict:
+            initial values in creating account.move.line
+        :param amount float:
+            amount derived from pos.payment, pos.order, or pos.order.line records
+        :param amount_converted float:
+            converted value of `amount` from the given `session_currency` to company currency
+        :param session_currency 'res.currency':
+            currency of the session (based on config_id)
+        :param is_using_company_currency bool:
+            whether pos session is using company currency (`session_currency` == company currency)
+
+        :return dict: complete values for creating 'amount.move.line' record
+        """
+        if is_using_company_currency:
+            if amount > 0:
+                return {'credit': amount, 'debit': 0.0, **partial_move_line_vals}
+            else:
+                return {'credit': 0.0, 'debit': abs(amount), **partial_move_line_vals}
+        else:
+            if amount_converted > 0:
+                init_vals = {'credit': amount_converted, 'debit': 0.0, **partial_move_line_vals}
+            else:
+                init_vals = {'credit': 0.0, 'debit': abs(amount_converted), **partial_move_line_vals}
+            return {
+                'amount_currency': -amount if amount_converted > 0 else amount,
+                'currency_id': session_currency.id,
+                **init_vals
+            }
+
+    @api.model
+    def _debit_amounts(self, partial_move_line_vals, amount, amount_converted, session_currency, is_using_company_currency):
+        """ `partial_move_line_vals` is completed by `debit`ing the given amounts.
+
+        See _credit_amounts docs for more details.
+        """
+        if is_using_company_currency:
+            if amount > 0:
+                return {'credit': 0.0, 'debit': amount, **partial_move_line_vals}
+            else:
+                return {'credit': abs(amount), 'debit': 0.0, **partial_move_line_vals}
+        else:
+            if amount_converted > 0:
+                init_vals = {'credit': 0.0, 'debit': amount_converted, **partial_move_line_vals}
+            else:
+                init_vals = {'credit': abs(amount_converted), 'debit': 0.0, **partial_move_line_vals}
+            return {
+                'amount_currency': amount if amount_converted > 0 else -amount,
```

Comment:
abs(amount)

Reply:
No. The sign is important for the `amount_currency` field.